tests/sentry/notifications/platform/slack/test_provider.py

```diff
@@ -56,3 +75,90 @@ def test_basic_fields(self) -> None:
     def test_is_available(self) -> None:
         assert SlackNotificationProvider.is_available() is False
         assert SlackNotificationProvider.is_available(organization=self.organization) is False
+
+
+class SlackNotificationProviderSendTest(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.integration, self.org_integration = self.create_provider_integration_for(
+            provider=IntegrationProviderSlug.SLACK,
+            organization=self.organization,
+            user=self.user,
+            name="test-slack",
+            metadata={"domain_name": "test-workspace.slack.com"},
+        )
+
+    def _create_target(self, resource_id: str = "C1234567890") -> IntegrationNotificationTarget:
+        target = IntegrationNotificationTarget(
+            provider_key=NotificationProviderKey.SLACK,
+            resource_id=resource_id,
+            resource_type=NotificationTargetResourceType.CHANNEL,
+            integration_id=self.integration.id,
+            organization_id=self.organization.id,
+        )
+        return target
+
+    def _create_renderable(self) -> SlackRenderable:
+        """Create a sample SlackRenderable for testing"""
+        from slack_sdk.models.blocks import (
+            HeaderBlock,
+            MarkdownTextObject,
+            PlainTextObject,
+            SectionBlock,
+        )
+
+        return SlackRenderable(
+            blocks=[
+                HeaderBlock(text=PlainTextObject(text="Test Notification")),
+                SectionBlock(text=MarkdownTextObject(text="This is a test message")),
+            ]
+        )
+
+    @patch("sentry.integrations.slack.integration.SlackSdkClient")
+    def test_send_success(self, mock_slack_client: Mock) -> None:
+        """Test successful message sending"""
+        mock_client_instance = mock_slack_client.return_value
+        mock_client_instance.chat_postMessage.return_value = {"ok": True, "ts": "1234567890.123456"}
+
+        target = self._create_target()
+        renderable = self._create_renderable()
+
+        SlackNotificationProvider.send(target=target, renderable=renderable)
+
+        mock_client_instance.chat_postMessage.assert_called_once_with(
+            channel="C1234567890", blocks=renderable["blocks"]
+        )
+
+    def test_send_invalid_target_class(self) -> None:
+        """Test send fails with invalid target class"""
+
+        target = GenericNotificationTarget(
+            provider_key=NotificationProviderKey.SLACK,
+            resource_id="test@example.com",
+            resource_type=NotificationTargetResourceType.EMAIL,
+        )
+        renderable = self._create_renderable()
+
+        with pytest.raises(NotificationProviderError, match="Target .* is not a valid dataclass"):
```

Comment:
While it's easy to use .* for this, should we be asserting that the notification message contains the correct string?